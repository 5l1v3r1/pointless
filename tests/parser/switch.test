
x = switch {
  case foo bar
}

x = switch {
  case foo bar
  case bar foo
}

x = switch {
  case foo bar
  otherwise foo
}

x = switch {
  case 1 == 2 bar
  case false a where a = 1
  otherwise foo where foo = 1
}