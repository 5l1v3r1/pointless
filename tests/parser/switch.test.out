(Program
 (Def
  (Name "x")
  (Requires
   (Name "bar")
   (Name "foo")))
 (Def
  (Name "x")
  (Conditional
   (Name "foo")
   (Name "bar")
   (Requires
    (Name "foo")
    (Name "bar"))))
 (Def
  (Name "x")
  (Conditional
   (Name "foo")
   (Name "bar")
   (Name "foo")))
 (Def
  (Name "x")
  (Conditional
   (BinaryOp
    Equals
    (Number 1)
    (Number 2))
   (Name "bar")
   (Conditional
    (Bool false)
    (Where
     (Name "a")
     (Object
      (Def
       (Name "a")
       (Number 1))))
    (Where
     (Name "foo")
     (Object
      (Def
       (Name "foo")
       (Number 1))))))))
